src/mesh.cpp

```diff
@@ -1,80 +1,116 @@
-{
-  if (!fs::is_regular_file(filename)) {
-    throw std::invalid_argument{"The mesh connectivity file does not exist: " + filename};
+  vtkSmartPointer<vtkGenericDataObjectReader> reader =
+      vtkSmartPointer<vtkGenericDataObjectReader>::New();
+  reader->SetFileName(filename.c_str());
+  reader->SetReadAllScalars(true);
+  reader->SetReadAllVectors(true);
+  reader->ReadAllFieldsOn();
+  reader->Update();
+
+  // Get Points
+  vtkPoints *Points = reader->GetUnstructuredGridOutput()->GetPoints();
+  vtkIdType  NumPoints =
+      reader->GetUnstructuredGridOutput()->GetNumberOfPoints();
+  double vertexPos[3];
+  for (vtkIdType point = 0; point < NumPoints; point++) {
+    Points->GetPoint(point, vertexPos);
+    std::array<double, 3> vertexPosArr{vertexPos[0], vertexPos[1],
+                                       vertexPos[2]};
+    mesh.positions.push_back(vertexPosArr);
   }
-  std::ifstream connFile{filename};
-  std::string   line;
-  while (std::getline(connFile, line)) {
-    std::vector<std::string> parts;
-    boost::split(parts, line, [](char c) { return c == ' '; });
-    std::vector<size_t> indices(parts.size());
-    std::transform(parts.begin(), parts.end(), indices.begin(), [](const std::string &s) -> size_t { return std::stol(s); });
-
-    if (indices.size() == 3) {
-      std::array<size_t, 3> elem{indices[0], indices[1], indices[2]};
-      mesh.triangles.push_back(elem);
-    } else if (indices.size() == 2) {
-      std::array<size_t, 2> elem{indices[0], indices[1]};
-      mesh.edges.push_back(elem);
-    } else {
-      throw std::runtime_error{std::string{"Invalid entry in connectivitiy file \""}.append(line).append("\"")};
+  // Get Point Data
+  vtkPointData *PD = reader->GetUnstructuredGridOutput()->GetPointData();
+  // Check it has data array
+  if (PD->HasArray(dataname.c_str()) == 1) {
+    // Get Data and Add to Mesh
+    vtkDataArray *ArrayData = PD->GetArray(dataname.c_str());
+    int           NumComp   = ArrayData->GetNumberOfComponents();
+    switch (NumComp) {
+    case 1: // Scalar Data
+      for (vtkIdType tupleIdx = 0; tupleIdx < NumPoints; tupleIdx++) {
+        const double scalar = ArrayData->GetTuple1(tupleIdx);
+        mesh.data.push_back(scalar);
+      }
+      break;
+    case 2: // Vector Data with 2 component
+      double *vector2ref;
+      for (vtkIdType tupleIdx = 0; tupleIdx < NumPoints; tupleIdx++) {
+        vector2ref = ArrayData->GetTuple2(tupleIdx);
+        mesh.data.push_back(vector2ref[0]);
+        mesh.data.push_back(vector2ref[1]);
+      }
+      break;
+    case 3: // Vector Data with 3 component
+      double *vector3ref;
+      for (vtkIdType tupleIdx = 0; tupleIdx < NumPoints; tupleIdx++) {
+        vector3ref = ArrayData->GetTuple3(tupleIdx);
+        mesh.data.push_back(vector3ref[0]);
+        mesh.data.push_back(vector3ref[1]);
+        mesh.data.push_back(vector3ref[2]);
+      }
+      break;
     }
+  } else { // There is no data in mesh file fill with zeros.
```

Comment:
Actually, I missed that part, you are right if there is more/less than the expected component we should have another error. Should I take the dimension of data from the user as a command-line input? Assuming that if data exist no input is needed but if data is missing somehow the number of components is required.

Reply:
_Hm_ I guess we cannot derive this information from the input arguments since you need to call `readBlockVectorData` or `readBlockScalarData`. Since we only need to distinguish these two options, you could just ask the user to provide the data type `vector` or `scalar`. You can then use `getDimensions` for the value.

Reply:
An input argument `--vector` is added now. If this flag is used, ASTE behaves data as a vector, if it is not used acts as scalar data if there is no input data found in VTK file.

Reply:
Alright. I think we need to document this later on, but the documentation is anyway deprecated at the moment so maybe just wait until we merged the other stuff.